src/renderer/components/dock/dock.tsx

```diff
@@ -46,19 +46,31 @@ interface Props {
 
 @observer
 export class Dock extends React.Component<Props> {
-  onKeydown = (evt: React.KeyboardEvent<HTMLElement>) => {
-    const { close, closeTab, selectedTab } = dockStore;
+  private element = React.createRef<HTMLDivElement>();
 
-    if (!selectedTab) return;
-    const { code, ctrlKey, shiftKey } = evt.nativeEvent;
+  componentDidMount() {
+    document.addEventListener("keydown", this.onKeyDown);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("keydown", this.onKeyDown);
+  }
+
+  onKeyDown = (evt: KeyboardEvent) => {
+    const { close, selectedTab, closeTab } = dockStore;
+    const { code, ctrlKey, metaKey, shiftKey } = evt;
+    // Determine if user working inside <Dock/> or using any other areas in app
+    const dockIsFocused = this.element?.current.contains(document.activeElement);
+
+    if (!selectedTab || !dockIsFocused) return;
```

Comment:
I found that `shift+cmd+w` is also very common keyboard shortcut among the apps. My guess it is used in the applications where some kind of tabs are presented. I would keep `shift+cmd+w` for closing window since it's not required to write extra code preventing triggering global shortcuts. And there might be situation when document focused element is unclear and attempt to close tab will lead to closing whole window as described in #3673 .

Reply:
Should we somehow indicate where the focus is? Maybe the color under the tab could be gray when not in focus and blue when it is? Or maybe the whole dock could have an outline when there's focus?

Reply:
Maybe this for consistency?

Windows: `Shift+Ctl+W` => exit app (suggested), `Ctrl+W` closing active tab (and currently whole window if tab is not focused?)
Mac: `Shift+Cmd+W` => exit app (like now), `Ctrl+W` closing active tab